packages/firestore/src/api/credentials.ts

```diff
@@ -240,35 +240,37 @@ export class FirebaseCredentialsProvider implements CredentialsProvider {
   }
 }
 
-// TODO(b/32935141): Ideally gapi type would be declared as an extern
-// tslint:disable-next-line:no-any
-export type Gapi = any;
+// Manual type definition for the subset of Gapi we use.
+interface Gapi {
+  auth: {
+    getAuthHeaderValueForFirstParty: (
+      userIdentifiers: Array<{ [key: string]: string }>
+    ) => string | null;
+  };
+}
 
 /*
  * FirstPartyToken provides a fresh token each time its value
  * is requested, because if the token is too old, requests will be rejected.
- * TODO(b/33147818) this implementation violates the current assumption that
- * tokens are immutable.  We need to either revisit this assumption or come
- * up with some way for FPA to use the listen/unlisten interface.
+ * Technically this may no longer be necessary since the SDK should gracefully
+ * recover from unauthenticated errors (see b/33147818 for context), but it's
+ * safer to keep the implementation as-is.
  */
 export class FirstPartyToken implements Token {
   type = 'FirstParty' as TokenType;
   user = User.FIRST_PARTY;
 
-  constructor(private gapi: Gapi, private sessionIndex: string) {
-    assert(
-      this.gapi &&
-        this.gapi['auth'] &&
-        this.gapi['auth']['getAuthHeaderValueForFirstParty'],
-      'unexpected gapi interface'
-    );
-  }
+  constructor(private gapi: Gapi, private sessionIndex: string) {}
 
   get authHeaders(): { [header: string]: string } {
-    return {
-      Authorization: this.gapi['auth']['getAuthHeaderValueForFirstParty']([]),
```

Comment:
getAuthHeaderValueForFirstParty() can technically return null, which would violate this method's return value.  So I tweaked this to omit 'Authorization' in that case.